velox/functions/lib/aggregates/noisy_aggregation/sketch/BitMap.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+#include "velox/common/memory/HashStringAllocator.h"
+
+namespace facebook::velox::functions::aggregate {
+
+/// A C++ implementation of the Bitmap class used in SfmSketch.
+/// This is a wrapper around std::vector<int8_t> which provides functionality
+/// similar to Java's BitSet. We can't use existing C++ implementations of
+/// bit_map using std::vector<int64_> because they are not compatible with
+/// Java's BitSet.toByteArray() which truncates trailing 0s in byte size.
+class BitMap {
+ public:
+  explicit BitMap(uint32_t length, HashStringAllocator* allocator);
```

Comment:
@oliver6782 Thank you for iterating.

I'm seeing that this new class combines a lot of unrelated functionality. I'd like to split it up.

Let's separate the logic of manipulating the bits from serde and memory allocation. 

Let's make sure to reuse existing terms: BitSet (instead of BitMap) and negate (instead of flip).

Let's make sure to clearly document all public methods (looking at flipBit templates specifically).

Let's make sure to following coding style. Please review at https://github.com/facebookincubator/velox/blob/main/CODING_STYLE.md Let us know if it is missing anything.

Ping me in the chat if you'd like to discuss more. Thanks.

CC: @amitkdutta